controllers/awscluster_controller.go

```diff
@@ -194,26 +232,37 @@ func reconcileDelete(clusterScope *scope.ClusterScope) (reconcile.Result, error)
 	// Cluster is deleted so remove the finalizer.
 	controllerutil.RemoveFinalizer(clusterScope.AWSCluster, infrav1.ClusterFinalizer)
```

Comment:
How can we find out these unrecoverable errors(like if the ownerRef is deleted manually)?

Reply:
I think it's difficult to track manual intervention, anybody can do kubectl edit and delete the finalizers and ownerRef without getting tracked, we can probably provide a warning somewhere to avoid such interventions.

Reply:
We should proceed on notFound error.

Reply:
I have pushed a new change to ignore deletion of finalizer and ownerRef from identity to delete ClusterFinalizer, please have a look.